### ISSUE TITLE
Eldoc output dumped to process buffer

### ISSUE THREAD

**Issue by ShuguangSun** [emacs-ess/ESS#794]
ess-20181206.2103 from elpa

It seems print some debug information in the inferior buffer as below. The line deparsing the input is inserted into the buffer.
```
> (list "" '(("..." . "")) '("recursive" "use.names" "..." "x"))
> (list "" '(("..." . "")) '("recursive" "use.names" "..." "x"))
> (list "" '(("..." . "")) '("recursive" "use.names" "..." "x"))
> c("RFSTDTC", "BRTHDTC")
```

**Comment by jabranham** [comment]
@vspinu I see this from time to time too. It's not during a long-taking task (i.e. R isn't busy). What's the best way to debug it?

**Comment by jabranham** [comment]
@vspinu I still see this frequently. Reopening.

**Comment by vspinu** [comment]
I don't see it at all. If this happens when R is not busy then it means there is a flaw in `ess-command` which shows on your machine. Most likely it doesn't detect the right end prompt. Maybe there is an extraneous ">" somewhere.

**Comment by jabranham** [comment]
That seems likely. Possibly related to or the same as #759. I experience both of these issues really frequently, it's quite annoying :-(

**Comment by vspinu** [comment]
What messages do you see? The OP messages are eldoc for `c(`. Normally arguments should be cached and never queried again, but it looks like something is not quite right with `c(` which is treated specially internally.

**Comment by jabranham** [comment]
I see it for a variety of functions. `c(`, `ungroup(` (from dplyr), `table` (from base). Here's the most recent two:

```R
> (list "dplyr" '(("x" . "") ("..." . "")) '("x" "..."))
> (list "base" '(("x" . "")) '("x"))
```

which I think were from ungroup and table. I see it happen both when navigating around an R script and also when typing at the prompt.

**Comment by vspinu** [comment]
How frequent is frequent? The thing is that I don't see this at all, and haven't seen before the recent changes except for rare occasions on long computation tasks. So it must be coming from the differences in our setups. I have switched to `nowait` for `ess-eval-visibly` which I believe it's your case, right?

**Comment by jabranham** [comment]
> How frequent is frequent?

It varies. Sometimes I go a while without seeing it, then it happens several times within a minute. 

> I have switched to nowait for ess-eval-visibly which I believe it's your case, right?

Yes, but I'm not sure it's related to `ess-eval-visibly`. I see it when I'm not evaluating something, just moving point around in the R script or even just typing at the prompt in the inferior buffer.

**Comment by jabranham** [comment]
Yes, they both show that. Just running it shows it in the inferior process and running the lisp shows it in a new buffer.

**Comment by vspinu** [comment]
I am afraid you are on your own on this one. Something is not right in the ess-command on your machine, it either blows or exits early. I would suggest you activate `ess-verbose` and place a few more `ess-if-verbose-write` in ess-command. Next time it happens just check the dribble buffer.

**Comment by slyrus** [comment]
I see this too from time to time, and AFAICT, this just started happening. I'll keep an eye out for it and see if I can get the actual inserted strings.

**Comment by vspinu** [comment]
I have a hunch. It's probably related to #804 and should be fixed in master.

**Comment by jabranham** [comment]
It's not fixed in master. I haven't had time to look into this yet, hopefully will be able to this week.

**Comment by vspinu** [comment]
Is this today's build? I have committed an improvement yesterday. I would be quite surprised if this behavior persists. You might be sending your own stuff to the sub-process in the background (timers, window change setup maybe?) which can cause such interference. Make sure it's on clean ESS that you see those.

**Comment by jabranham** [comment]
> Is this today's build?

Yes

OK, I think I've chased down the problem. If you set `ess-eldoc-show-on-symbol` to `t` and `eldoc-delay` to `0`, then you (sometimes) experience this.

**Comment by vspinu** [comment]
I think it's not a bad upstream change but it's certainly a bad change for us. 

One workaround would be to place the eldoc message inside special markers (something that is currently used as part of our rudimentary MPI. Actually eldoc can just be an MPI message I believe. 

We might try that first and then ask upstream if it doesn't work. Unfortunately I am swamped with work this week. When is the release planned?

**Comment by jabranham** [comment]
This change will first appear in Emacs 27.1, which won't be out for a while yet. Probably late this year at the absolute earliest. I'd bet next year sometime though. So it's not a big rush.

**Comment by jabranham** [comment]
@vspinu This one really needs to get tackled before the 19.04 release. Can you give an example of where/how the mpi stuff is used? I can't figure out exactly how it works.

**Comment by vspinu** [comment]
No time to dwell into this ATM, sorry. 

We can postpone the release if required. I haven't had a good chance to test ESS with all the new stuff.

**Comment by mguzmann** [comment]
I currently have this issue, any workarounds?

**Comment by djh1202** [comment]
I'm also having this exact issue on GNU Emacs 27.1 using ess-18.10.2 from melpa-stable.

The issue seems to disappear if I use the MELPA version of ess, but that version doesn't support stata mode, which I need. 

Any ideas? Is there some way to keep eldoc functionality for R without losing stata mode?

**Comment by markseeto** [comment]
> Actually all that's necessary is `(setq eldoc-idle-delay 0)`.

@jabranham does that mean that having a non-zero value for `eldoc-idle-delay` should stop this problem from happening?

I recently started seeing this problem after changing from Emacs 25.2 (ESS 16.10) to Emacs 27.1 (ESS 18.10.2), both Vincent Goulet's distributions for Windows.

But in Emacs 27.1 when I do `C-h v eldoc-idle-delay` it says
```
Its value is 0.1
Original value was 0.5
Local in buffer test01.R; global value is 0.5
```
So I have the problem when `eldoc-idle-delay` is non-zero. I tried adding `(setq eldoc-idle-delay 5)` to my .emacs file and the problem still happens, although it still says its value is 0.1 with global value of 5.

Is there a workaround available to stop the problem without disabling eldoc altogether?

Thanks.
